dd-sdk-android-ndk/src/main/cpp/utils/signal-monitor.c

```diff
@@ -152,61 +146,127 @@ bool configure_signal_stack() {
     return true;
 }
 
-void invoke_previous_handler(int signum, siginfo_t *info,
-                             void *user_context) {
 
-    pthread_mutex_lock(&handlerMutex);
+bool configure_global_sigaction() {
+    global_sigaction = calloc(handled_signals_size(), sizeof(struct sigaction));
+    if (global_sigaction == NULL) {
+        return false;
+    }
+    sigemptyset(&global_sigaction->sa_mask);
+    global_sigaction->sa_sigaction = handle_signal;
+    // we will use the SA_ONSTACK mask here to handle the signal in a freshly new stack.
+    global_sigaction->sa_flags = SA_SIGINFO | SA_ONSTACK;
 
+    return true;
+}
+
+void *override_native_signal_handlers() {
+    pthread_mutex_lock(&mutex);
+    if (handlers_installed) {
+        pthread_mutex_unlock(&mutex);
+        pthread_exit(NULL);
```

Comment:
Why do you kill this thread? Didn't you just want to return here?

Reply:
Wanted to make sure it is killed but yes you are right I can just return :)